src/data_hub/dataframe.py

```diff
@@ -0,0 +1,159 @@
+
+    The primary method of creating an object of this type is to use the [api.get_dataframe][src.data_hub.api.get_dataframe] function.
+    """
+
+    auto_sync: bool = False
+    """When True, changes made to the dataframe will be automatically synced to the Deep Origin database this dataframe represents."""
+
+    class AtIndexer:
+        """this class override is used to intercept calls to at indexer of a pandas dataframe"""
+
+        def __init__(self, obj):
+            self.obj = obj
+
+        def __getitem__(self, key):
+            """intercept for the set operation"""
+
+            return self.obj._get_value(*key)
+
+        def __setitem__(self, key, value):
+            """intercept for the set operation""" ""
+
+            old_value = self.obj._get_value(*key)
+            if value == old_value:
+                # noop
+                return
+
+            rows = [key[0]]
+            columns = [key[1]]
+
+            # Perform the actual setting operation
+            self.obj._set_value(*key, value)
+
+            # now update the DB. note that self is an AtIndexer
+            # object, so we need to index into the pandas object
+            if self.obj.auto_sync:
+                self.obj.sync(columns=columns, rows=rows)
+
+    @property
+    def at(self):
+        """Override the `at` property to return an AtIndexer"""
+        return self.AtIndexer(self)
+
+    def __setitem__(self, key, value):
+        """Override the __setitem__ method to update the Deep Origin database when changes are made to the local
+        dataframe"""
+
+        # first, call the pandas method
+        super().__setitem__(key, value)
+
+        # now, update the Deep Origin database with the changes
+        # we just made
+        if self.auto_sync:
+            self.sync(columns=[key])
+
+    def _repr_html_(self):
+        """method override to customize printing in a Jupyter notebook"""
+
+        name = self.attrs["metadata"]["name"]
+        url = construct_resource_url(
+            name=name,
+            row_type="database",
+        )
+
+        # Convert the string to a datetime object
+        date_str = self.attrs["metadata"]["dateCreated"]
+        date_obj = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f")
+
+        now = datetime.now()
+
+        # Calculate the difference
+        time_diff = now - date_obj
+
+        # Convert the time difference into "x time ago" format
+        time_ago = humanize.naturaltime(time_diff)
+
+        header = f'<h4>{name} <a href = "{url}">🔗</a></h4>'
+        txt = f'<p style="font-size: 12px; color: #808080;">Created {time_ago}.</p>'
+        df_html = super()._repr_html_()
+        return header + txt + df_html
+
+    def __repr__(self):
+        """method override to customize printing in an interactive session"""
+
+        header = f'{self.attrs["metadata"]["hid"]}\n'
+        df_representation = super().__repr__()
+        return header + df_representation
+
+    def sync(
```

Comment:
For consistency with pandas, I think we could rename this to `to_deeporigin`

Reply:
fixed